magic.lock

```diff
@@ -119,6 +119,7 @@ environments:
       - conda: https://conda.anaconda.org/conda-forge/linux-64/libxcrypt-4.4.36-hd590300_1.conda
       - conda: https://conda.anaconda.org/conda-forge/linux-64/libxml2-2.13.4-h064dc61_2.conda
       - conda: https://conda.anaconda.org/conda-forge/linux-64/libzlib-1.3.1-hb9d3cd8_2.conda
+      - conda: https://conda.anaconda.org/conda-forge/noarch/lit-19.1.3-pyhd8ed1ab_0.conda
```

Comment:
@Ahajha can you comment here how we're managing the magic lock files here internally wrt this?  Given the migration to magic, I agree with @thatstoasty that we could just specify this as part of the dependencies in the `pixi.toml`.

Reply:
I think this should be fine. 2 points:
1: This would allow a follow-up cleanup where we could remove the section of CI that installs `lit` through llvm.sh.
2: Internally, the lockfiles are wiped and then recreated each night (rather than updated in-place), so the only reason to update them here is to satisfy the `--frozen` flag, as `pixi.toml` is essentially ignored in CI here. (There might be a different flag that might be better for this use case, where it will check both?) After a night this should sync internally + regenerate, and then everything should be fine.

Verbal approval, from the CI side of things this should be a-ok.

@thatstoasty Thanks for improving the state of the world :)

Reply:
Great! I handle a lot of the CI for my teams at work, so I know how fickle internal processes can be 🙂. Glad to see that the change might make its way downstream to make contributing a little easier!